src/android/java/io/jxcore/node/BtConnectorHelper.java

```diff
@@ -232,18 +241,21 @@ public void BeginConnectPeer(String toPeerId, ConnectStatusCallback connectStatu
         }
     }
 
+    //this is always called in context of thread that created instance of the library
     @Override
     public void Connected(BluetoothSocket bluetoothSocket, boolean incoming,String peerId,String peerName,String peerAddress) {
 
         if(bluetoothSocket != null) {
 
             AddPeerIfNotDiscovered(bluetoothSocket, peerId, peerName, peerAddress);
-            print_debug("Starting the connected thread incoming : " + incoming);
+            print_debug("Starting the connected thread incoming : " + incoming + ", " +peerName);
```

Comment:
@DrJukka This comment is in regards to line 239 above where you return "Device Address for .... not found from Discovered device list". The code, I believe, would be easier to read if you instead put in

```java
if (selectedDevice == null) {
  tmpCallback.ConnectionStatusUpdate("Device Address for " + toPeerId + " not found from Discovered device list.",-1);
  return;
}
```
Then you can remove the `if (selectedDevice != null)` and out indent that code.